frontend/src/libs/packages/socket/socket.package.ts

```diff
@@ -1,38 +1,64 @@
 import { type Socket, io } from 'socket.io-client';
 
 import { config } from '~/libs/packages/config/config.js';
+import {
+  type UserEntityObjectWithGroupT,
+  type ValueOf,
+} from '~/libs/types/types.js';
 
+import { type SocketResponseStatus } from './libs/enums/enums.js';
 import {
-  type ClientSocketEventValue,
+  type ClientSocketEventParameter,
   type ServerSocketEventParameter,
+  type ServerSocketEventResponse,
 } from './libs/types/types.js';
 
 class SocketService {
   private io: Socket | undefined;
 
-  public connect(): void {
+  public connect(user: UserEntityObjectWithGroupT | null): void {
+    const auth = user ? { userId: user.id } : {};
     this.io = io(config.ENV.API.SERVER_URL, {
       transports: ['websocket', 'polling'],
+      auth,
     });
   }
 
```

Comment:
```
type ServerToClientEvents = {
  noArg: () => void;
  basicEmit: (a: number, b: string, c: Buffer) => void;
  withAck: (d: string, callback: (error: number) => void) => void;
};

type ClientToServerEvents = {
  hello: () => void;
};

class SocketService {
  private io: Socket<ServerToClientEvents, ClientToServerEvents> | undefined;

  public connect(): void {
    const socket: Socket<ServerToClientEvents, ClientToServerEvents> = io(
      config.ENV.API.SERVER_URL,
      {
        transports: ['websocket', 'polling'],
      },
    );
    this.io = socket;
  }
```
Maybe we can pass ServerToClientEvents and ClientToServerEvents as generics like this (like in socket.io docs)

Reply:
Also let's make it in another ticket to merge the main functionality sooner.